bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py

```diff
@@ -221,19 +226,242 @@ def predict(self):
+                    w = int(w * original_image_width)
+                    x = x * original_image_width - w / 2
+                    y = y * original_image_height - h / 2
+                    list_of_coordinates = [int(x), int(y), int(w), int(h)]
+                    # Convert to int
+                    objects.append([list_of_coordinates, float(confidence), j])
+        return objects
+
+    def predict(self):
+        if (self._ball_candidates is None and self._goalpost_candidates is None) or not self._caching:
+            # Set up variables
+            self._ball_candidates = list()
+            self._goalpost_candidates = list()
+
+            rospy.logdebug("Starting inference...", logger_name="vision_yolo")
+
+            # Set request id for the stick. Since we only make one call at a time, we use a static parameter.
+            request_id = 1
+            # Resize image to yolo input size
+            in_frame = cv2.resize(self._image, (self._w, self._h))
+
+            # resize input_frame to network size
+            in_frame = in_frame.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+            in_frame = in_frame.reshape((self._n, self._c, self._h, self._w))
+
+            # Start inference
+            self._exec_net.start_async(request_id=request_id, inputs={self._input_blob: in_frame})
+
+            # Collecting object detection results
+            detections = list()
+            # Create barrier. This lets all following processing steps wait until the prediction is calculated.
+            if self._exec_net.requests[request_id].wait(-1) == 0:
+                # Get output
+                output = self._exec_net.requests[request_id].outputs
+                # Iterate over output layers
+                for layer_name, out_blob in output.items():
+                    # Reshape output layer
+                    out_blob = out_blob.reshape(self._net.layers[self._net.layers[layer_name].parents[0]].shape)
+                    # Create layer params object
+                    layer_params = self._YoloParams(self._net.layers[layer_name].params, out_blob.shape[2])
+                    # Parse yolo bounding boxes out of output blob
+                    detections.extend(
+                        self._parse_yolo_region(
+                            out_blob,
+                            in_frame.shape[2:],
+                            self._image.shape[:-1],
+                            layer_params,
+                            self._confidence_threshold))
+
+            if detections:
+                # Transpose detections
+                boxes, confidences, class_ids = list(map(list, zip(*detections)))
+                # Non-maximum Suppression
+                box_indices = cv2.dnn.NMSBoxes(boxes, confidences, self._confidence_threshold, self._nms_threshold)
+                # Iterate over filtered boxes
+                for index in box_indices:
+                    # Get id
```

Comment:
```suggestion
                    # Get class id of detection
```

Reply:
This is not correct. It is the Id of the detection itself, not it's class.